### PR TITLE
Ignore synthesized delegates in AnonymousTypeManager.GetAnonymousTypeMap

### PR DESCRIPTION
**Pull request by cston**
Ignore synthesized delegates in AnonymousTypeManager.GetAnonymousTypeMap
and add generation qualifier to synthesized delegate names to avoid name clashes across EnC generations.

### CODE REVIEW

**Comment by cston** [comment]
@tmat, @dotnet/roslyn-compiler

**Comment by AlekseyTs** [comment]
Is this fixing a bug or an issue?

File: src/Compilers/CSharp/Portable/Symbols/AnonymousTypes/AnonymousTypeManager.Templates.cs [hunk 1]
```diff
@@ -405,14 +380,18 @@ internal static Microsoft.CodeAnalysis.Emit.AnonymousTypeKey GetAnonymousTypeKey
         internal IReadOnlyDictionary<Microsoft.CodeAnalysis.Emit.AnonymousTypeKey, Microsoft.CodeAnalysis.Emit.AnonymousTypeValue> GetAnonymousTypeMap()
         {
             var result = new Dictionary<Microsoft.CodeAnalysis.Emit.AnonymousTypeKey, Microsoft.CodeAnalysis.Emit.AnonymousTypeValue>();
-            var templates = GetAllCreatedTemplates();
-            foreach (AnonymousTypeTemplateSymbol template in templates)
+            var templates = ArrayBuilder<AnonymousTypeTemplateSymbol>.GetInstance();
+            // Get anonymous types but not synthesized delegates
+            // since delegate types are not reused across generations.
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Symbols/AnonymousTypes/AnonymousTypeManager.Templates.cs]: Is there something wrong in reusing the delegate types?

Comment (cston) [inline src/Compilers/CSharp/Portable/Symbols/AnonymousTypes/AnonymousTypeManager.Templates.cs]: No, it's just additional complexity. To reuse delegates from the initial generation (loaded from metadata) we'd need to parse the mangled delegate metadata names for instance.

**Comment by AlekseyTs** [comment]
LGTM

**Pull request merged**